src/panels/lovelace/common/directives/long-press-directive.ts

```diff
@@ -64,34 +65,59 @@ class LongPress extends HTMLElement implements LongPress {
       return;
     }
     element.longPress = true;
-    element.addEventListener(
-      isTouch ? "touchstart" : "mousedown",
-      (ev: Event) => {
-        this.held = false;
-        let x;
-        let y;
-        if ((ev as TouchEvent).touches) {
-          x = (ev as TouchEvent).touches[0].pageX;
-          y = (ev as TouchEvent).touches[0].pageY;
-        } else {
-          x = (ev as MouseEvent).pageX;
-          y = (ev as MouseEvent).pageY;
-        }
-        this.timer = window.setTimeout(() => {
-          this.startAnimation(x, y);
-          this.held = true;
-        }, this.holdTime);
-      },
-      { passive: true }
-    );
-    element.addEventListener("click", () => {
+
+    element.addEventListener("contextmenu", (ev: Event) => {
+      const e = ev || window.event;
+      if (e.preventDefault) {
+        e.preventDefault();
+      }
+      if (e.stopPropagation) {
+        e.stopPropagation();
+      }
+      e.cancelBubble = true;
+      e.returnValue = false;
+      return false;
+    });
+
+    const clickStart = (ev: Event) => {
+      this.held = false;
+      let x;
+      let y;
+      if ((ev as TouchEvent).touches) {
+        x = (ev as TouchEvent).touches[0].pageX;
+        y = (ev as TouchEvent).touches[0].pageY;
+      } else {
+        x = (ev as MouseEvent).pageX;
+        y = (ev as MouseEvent).pageY;
+      }
+      this.timer = window.setTimeout(() => {
+        this.startAnimation(x, y);
+        this.held = true;
+      }, this.holdTime);
+    };
+
+    const clickEnd = () => {
+      clearTimeout(this.timer);
       this.stopAnimation();
+      if (isTouch && this.timer == null) {
```

Comment:
you're checking if `timer == null` but `timer` is never set to `null`, just `undefined`. 

With just `==`, a comparison with `null` also matches `undefined`, but if we really only expect `undefined`, we should check for that.

![image](https://user-images.githubusercontent.com/1444314/48252644-9163b400-e405-11e8-9680-4f15b614acb7.png)